src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,157 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        checkSize();
+        array[size - 1] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index can't be less than zero");
+        } else if (index < size) {
+            size++;
+            T[] tempArray = (T[]) new Object[checkLength()];
+            for (int i = 0; i < index; i++) {
+                tempArray[i] = array[i];
+            }
+            if (index < size) {
+                for (int i = index + 1; i < tempArray.length; i++) {
+                    tempArray[i] = array[i - 1];
+                }
+            }
```

Comment:
delete for loops
There is **only one** place where you can have for loop in this task - public T remove(T element) 
![image](https://github.com/mate-academy/jv-array-list/assets/20326858/3dd3f545-cd01-45b1-99d0-6896e2491b99)